hil/client/node.py

```diff
@@ -32,7 +32,24 @@ def register(self, node, subtype, *args):
         # and currently active drivers for HIL
         # obm_api = "http://schema.massopencloud.org/haas/v0/obm/"
         # obm_types = ["ipmi", "mock"]
-        raise NotImplementedError
+        obm_api = "http://schema.massopencloud.org/haas/v0/obm/"
+        ext_url = self.object_url('active_extensions')
+        ext_resp = self.check_response(self.httpClient.request("GET", ext_url))
+        obm_types = []
+        for resp in ext_resp:
+            if "hil.ext.obm" in resp:
+                obm_types.append(resp.split(".")[-1])
+
+        if subtype in obm_types:
```

Comment:
you haven't handled the `else` case here. What if it's not in the subtype, we would still make the API call with nothing in the payload.

Reply:
Currently, I got `Error: Request arguments are not valid for this request`.  So should I add some error handling here?

Reply:
I think we could raise an exception when we detect that `subtype is not in obm_types`. Client library raises the `FailedAPICallException` with some helpful message.

Reply:
`FailedAPICallException` is a bit weird here since it's not in response to an error from the server. It might make more sense to raise a different error, though I don't feel super strongly.

Reply:
oh right, in that case you could define a new error in base.py and raise that.